sql/core.go

```diff
@@ -149,6 +149,14 @@ type Databaser interface {
 	WithDatabase(Database) (Node, error)
 }
 
+type SchemeModifiable interface {
+	// Schema of the node.
+	Schema() Schema
+	// WithSchema returns a new node instance with the schema replaced with
```

Comment:
For reasons discussed above, I have a suspicion that you have simply recreated a more complicated version of sql.Expressioner. At best you're going to need to plug this into many more places in the analyzer as time goes on. At worst this is just intractable. Spend some time seeing if you can get everything to work the same with sql.Expressioner and your life will get simpler.